core/common/src/main/java/alluxio/underfs/Fingerprint.java

```diff
@@ -135,6 +140,36 @@ public String serialize() {
     return sb.toString();
   }
 
+  /**
+   * Returns true if the serialized fingerprint matches the fingerprint in metadata.
+   *
+   * @param fp a parsed fingerprint object
+   * @return true if the given fingerprint matches the current fingerprint in metadata
+   */
+  public boolean matchMetadata(Fingerprint fp) {
+    for (Tag tag : METADATA_TAGS) {
+      if (!getTag(tag).equals(fp.getTag(tag))) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns true if the serialized fingerprint matches the fingerprint in the data part.
```

Comment:
NIT: I'd rather say "content" instead of "data". This applies to the comment below too.

Reply:
done

Reply:
can you change this line "data part" too?